include/revng/Support/OpaqueFunctionsPool.h

```diff
@@ -94,4 +95,29 @@ class OpaqueFunctionsPool {
 
     return get(Key, FunctionType::get(ReturnType, Arguments, false), Name);
   }
+
+  /// Initialize the pool with all the functions in M that match the tag TheTag,
+  /// using the return type as key.
+  void initializeFromReturnType(const FunctionTags::Tag &TheTag) requires
```

Comment:
The function name does not reflect the fact that it works with tags only.
Consider accepting a `Function *` range instead or having a more descriptive name.
No biggie though, it can also stay like this.

Reply:
>  Consider accepting a Function * range instead or having a more descriptive name.
I don't understand what you mean.

Reply:
Something like:
```c++
   void initializeFromReturnType(auto X) requires ... {
     for (llvm::Function &F : X) {
     }
   }
```
Call as:
```c++
initializeFromReturnType(TheTag.functions(M));
```
Maybe instead of `auto` above you can use `range` from `Concepts.h`.

Reply:
So you want the API to ignore the `Tag` and just get a list of `Function *`?

Reply:
Honestly, not taking a list of `Function *` was the whole point of this API: relieving the user from the task of manually collecting the function list manually. That would be a totally different API that we can still add in another moment, if someone wants it.

I'd go for renaming this API to incorporate tag:
- `initializeFromTaggedWithReturnTypeKey`
- `initializeFromTaggedWithNthArgTypeKey`

The naming is very verbose though. Do you have better suggestions? Can't we just be happy with the fact that there's documentation and the only argument is a `Tag`? What else should the API use?

Reply:
OK, no biggie.

Reply:
I'm confused now. Do you want me to change the name or just leave it like it is?

Reply:
I marked the issue as solved. No further changes are required.